azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -266,14 +318,137 @@ def client_add(
             resource_group=resource_group,
             spc_name=OPCUA_SPC_NAME,
             secret_sync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+            should_replace=True,
         )
 
         # update opcua extension
         return self._update_client_secret_to_extension(
+            applicationCert=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
             subject_name=subject_name,
             application_uri=application_uri,
         )
 
+    def remove(
+        self,
+        instance_name: str,
+        resource_group: str,
+        secretsync_name: str,
+        certificate_names: List[str],
+        confirm_yes: Optional[bool] = False,
+        force: Optional[bool] = False,
+        include_secrets: Optional[bool] = False,
+    ) -> dict:
+        # prompt for deletion
+        should_bail = not should_continue_prompt(confirm_yes=confirm_yes)
+        if should_bail:
+            return
+
+        if not force:
+            if not self.resource_map.connected_cluster.connected:
+                logger.warning(
+                    "Removal cancelled. The cluster is not connected to Azure. "
+                    "Use --force to continue anyway, which may lead to errors."
+                )
+                return
+
+        cl_resources = self._get_cl_resources(instance_name=instance_name, resource_group=resource_group)
```

Comment:
We should revisit self.get_cl_resources and self.instances.find_existing_resources

Reply:
adding a todo